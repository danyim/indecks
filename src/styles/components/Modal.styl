@import '../_variables';

.modal-overlay
  background-color: rgba(0, 0, 0, .15)
  position: fixed
  top: 0
  left: 0
  right: 0
  bottom: 0
  z-index: 1500

.modal-content
  width: 75%
  max-width: 700px
  background-color: white
  position: absolute
  top: 50%
  left: 50%
  right: auto
  bottom: auto
  margin-right: -50%
  transform: translate(-50%, -50%) !important
  border: 1px solid black
  padding: 3.5rem
  transition: all .2s

  :global
    .close-modal
      position: absolute
      right: 3.5rem
      top: 4.7rem
      cursor: pointer
      color: black
      z-index: 100
      // padding: 0.2rem
      // line-height: 1rem
      // width: 2rem
      // height: 2rem
      // text-align: center
      // border: 1px solid black
  h2
    margin-top: 0
    padding-bottom: .3rem
    border-bottom: .6rem solid black

@media (max-width: cardWidth)
  .modal-content
    width: 95%
    position: inherit
    top: 2%
    left: 50%
    transform: translate(-50%, 1%) !important
    padding: 2rem
    overflow: scroll
    // height: 95%
