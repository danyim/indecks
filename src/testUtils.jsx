/*
  Not sure if we want to use this yet in our tests...
  TODO: Find a definitive structure for Jest tests and follow it.
 */
import React from 'react';
import { shallow, mount } from 'enzyme';

export const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />)

  return {
    props,
    wrapper
  }
};

export const setupFull = (Component, props) => {
  const wrapper = mount(<Component {...props} />)

  return {
    props,
    wrapper
  }
};
