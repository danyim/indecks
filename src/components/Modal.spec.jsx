import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow, mount, setup, setupFull } from '../testUtils' // eslint-disable-line no-unused-vars
import Modal from './Modal'

const defaultProps = {
  onRequestClose: () => {},
  isOpen: false,
  children: <div />,
  contentLabel: 'Test modal',
  style: null,
  onAfterOpen: null,
  className: null,
  overlayClassName: null
}

describe('Modal', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(Modal, defaultProps)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should close the modal when close is clicked', () => {
    const handler = jest.fn()
    const { wrapper } = setup(Modal, {
      ...defaultProps,
      onRequestClose: handler
    })

    const closeButton = wrapper.find('a.close-modal')
    closeButton.simulate('click')
    expect(handler.mock.calls.length).toEqual(1)
  })

  it('should render the contents of its children', () => {
    const testDOM = <div className="test-dom">Test</div>
    const { wrapper } = setup(Modal, {
      ...defaultProps,
      children: testDOM
    })

    const dom = wrapper.find('div.test-dom')
    expect(dom.exists()).toBe(true)
  })
})
