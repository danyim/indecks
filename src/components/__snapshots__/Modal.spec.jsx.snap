// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render self and subcomponents 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal-content null"
  closeTimeoutMS={0}
  contentLabel="Test modal"
  isOpen={false}
  onAfterOpen={null}
  onRequestClose={[Function]}
  overlayClassName="modal-overlay null"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": null,
    }
  }
>
  <a
    className="close-modal"
    onClick={[Function]}
    role="presentation"
  >
    Close
  </a>
  <div />
</Modal>
`
